packages/context/test/unit/reflect.ts

```diff
@@ -0,0 +1,728 @@
+    it('lists metadata keys of class methods', () => {
+      const metadataValue: Object = {value: 'sample'};
+
+      reflectContext.defineMetadata('key3', metadataValue, SubClass, 'staticSubMethod');
+      reflectContext.defineMetadata('key4', metadataValue, BaseClass, 'staticBaseMethod');
+
+      reflectContext.defineMetadata('key5', metadataValue, SubClass.prototype, 'subMethod');
+      reflectContext.defineMetadata('key6', metadataValue, SubClass.prototype, 'baseMethod');
+      reflectContext.defineMetadata('key7', metadataValue, BaseClass.prototype, 'baseMethod');
+
+      let keys = reflectContext.getOwnMetadataKeys(SubClass, 'staticSubMethod');
+      expect(keys).to.eql(['key3']);
+
+      keys = reflectContext.getOwnMetadataKeys(SubClass, 'staticBaseMethod');
+      expect(keys).to.eql([]);
+
+      keys = reflectContext.getOwnMetadataKeys(BaseClass, 'staticBaseMethod');
+      expect(keys).to.eql(['key4']);
+
+      keys = reflectContext.getOwnMetadataKeys(SubClass.prototype, 'subMethod');
+      expect(keys).to.eql(['key5']);
+
+      keys = reflectContext.getOwnMetadataKeys(SubClass.prototype, 'baseMethod');
+      expect(keys).to.eql(['key6']);
+
+      keys = reflectContext.getMetadataKeys(SubClass.prototype, 'baseMethod');
+      expect(keys).to.eql(['key6', 'key7']);
+
+      keys = reflectContext.getOwnMetadataKeys(BaseClass.prototype, 'baseMethod');
+      expect(keys).to.eql(['key7']);
+    });
+
+
+    it('checks hasMetadata against a class', () => {
+      const metadataValue: Object = {value: 'sample'};
+      // define a metadata using the namespaced reflectContext
+      reflectContext.defineMetadata('key1', metadataValue, SubClass);
+      reflectContext.defineMetadata('key2', {}, BaseClass);
+
+      let result = reflectContext.hasMetadata('key1', SubClass);
+      expect(result).to.be.true();
+
+      result = reflectContext.hasMetadata('key2', SubClass);
+      expect(result).to.be.true();
+
+      result = reflectContext.hasMetadata('key1', BaseClass);
+      expect(result).to.be.false();
+
+      result = reflectContext.hasMetadata('key2', BaseClass);
+      expect(result).to.be.true();
+    });
+
+    it('checks hasOwnMetadata against a class', () => {
+      const metadataValue: Object = {value: 'sample'};
+      // define a metadata using the namespaced reflectContext
+      reflectContext.defineMetadata('key1', metadataValue, SubClass);
+      reflectContext.defineMetadata('key2', {}, BaseClass);
+
+      let result = reflectContext.hasOwnMetadata('key1', SubClass);
+      expect(result).to.be.true();
+
+      result = reflectContext.hasOwnMetadata('key2', SubClass);
+      expect(result).to.be.false();
+
+      result = reflectContext.hasOwnMetadata('key1', BaseClass);
+      expect(result).to.be.false();
+
+      result = reflectContext.hasOwnMetadata('key2', BaseClass);
+      expect(result).to.be.true();
+    });
+
+  });
+
+  describe('default Reflector', () => {
+    beforeEach(givenReflectContext);
```

Comment:
do we need to add all test cases for the default Reflector ? wont it be enough to check the type of the default reflector ?